expression/builtin_vectorized.go

```diff
@@ -32,17 +32,20 @@ type columnBufferAllocator interface {
 }
 
 // localSliceBuffer implements columnBufferAllocator interface.
-// It works like a concurrency-safe deque which is implemented by a lock + slice.
+// It works like a concurrency-safe deque which is implemented by sync.Pool.
 type localSliceBuffer struct {
-	sync.Mutex
-	buffers []*chunk.Column
-	head    int
-	tail    int
-	size    int
+	pool sync.Pool
 }
 
 func newLocalSliceBuffer(initCap int) *localSliceBuffer {
-	return &localSliceBuffer{buffers: make([]*chunk.Column, initCap)}
+	return &localSliceBuffer{
+		pool: sync.Pool{
+			New: func() interface{} {
+				// Use default arguments
+				return chunk.NewColumn(types.NewFieldType(mysql.TypeLonglong), chunk.InitialCapacity)
```

Comment:
Why using `chunk.InitialCapacity` instead of `initCap`?

Reply:
Because initCap here means the size of buffer instead of the size of chunk.Column.
So I choose to use a default value.

Reply:
OK, it seems that we can delete this argument of `newLocalColumnPool`.